v0/destinations/profitwell/transform.js

```diff
@@ -0,0 +1,294 @@
+      // dropping event if profitwellSubscriptionId (externalId) did not
+      // match with any subscription_id at destination
+      if (subscriptionId) {
+        throw new CustomError("profitwell subscription_id not found", 400);
+      }
+      payload = constructPayload(message, createPayloadMapping);
+      payload = {
+        ...payload,
+        user_id: userId,
+        user_alias: userAlias
+      };
+      if (
+        payload.plan_interval &&
+        !(
+          payload.plan_interval.toLowerCase() === "month" ||
+          payload.plan_interval.toLowerCase() === "year"
+        )
+      ) {
+        throw new CustomError("invalid format for planInterval. Aborting", 400);
+      }
+      if (
+        payload.plan_currency &&
+        !isValidPlanCurrency(payload.plan_currency)
+      ) {
+        payload.plan_currency = null;
+      }
+      if (
+        payload.status &&
+        !(
+          payload.status.toLowerCase() === "active" ||
+          payload.status.toLowerCase() === "trialing"
+        )
+      ) {
+        payload.status = null;
+      }
+      payload.effective_date = unixTimestampOrError(
+        payload.effective_date,
+        message.originalTimestamp
+      );
+      response.method = defaultPostRequestConfig.requestMethod;
+      response.endpoint = `${BASE_ENDPOINT}/v2/subscriptions/`;
+      response.headers = {
+        "Content-Type": "application/json",
+        Authorization: Config.privateApiKey
+      };
+      response.body.JSON = removeUndefinedAndNullValues(payload);
+      return response;
+    }
+
+    // for a given userId, subscription is found at dest.
+    if (valFound) {
+      payload = constructPayload(message, updatePayloadMapping);
+      if (
+        payload.plan_interval &&
+        !(
+          payload.plan_interval.toLowerCase() === "month" ||
+          payload.plan_interval.toLowerCase() === "year"
+        )
+      ) {
+        throw new CustomError("invalid format for planInterval. Aborting", 400);
+      }
+      if (
+        payload.status &&
+        !(
+          payload.status.toLowerCase() === "active" ||
+          payload.status.toLowerCase() === "trialing"
+        )
+      ) {
+        payload.status = null;
+      }
+      payload.effective_date = unixTimestampOrError(
+        payload.effective_date,
+        message.originalTimestamp
+      );
+      response.method = defaultPutRequestConfig.requestMethod;
+      response.endpoint = `${BASE_ENDPOINT}/v2/subscriptions/${subscriptionId ||
+        subscriptionAlias}/`;
+      response.headers = {
+        "Content-Type": "application/json",
+        Authorization: Config.privateApiKey
+      };
+      response.body.JSON = removeUndefinedAndNullValues(payload);
+      return response;
+    }
+  }
+
+  // handler for other destination side errors
+  if (res.response.response.status !== 404) {
+    throw new CustomError(
+      "Failed to get subscription history for a user",
```

Comment:
How about replacing the error message with destination response as `error.response.statusText` to make the error message more clear as it can return `401` too for unauthorized access while providing invalid key/credentials.

Reply:
yes you can maybe append the statusText at the end of the error message that is already there. should be good enough.

Reply:
append to the end seems good.